core/metrics/src/main/scala/org/typelevel/otel4s/metrics/AsyncInstrumentBuilder.scala

```diff
@@ -0,0 +1,13 @@
+package org.typelevel.otel4s.metrics
+
+import cats.effect.Resource
+
+trait AsyncInstrumentBuilder[F[_], Input, Instrument[_[_], _]] {
+  type Self <: AsyncInstrumentBuilder[F, Input, Instrument]
+
+  def withUnit(unit: String): Self
+  def withDescription(description: String): Self
+  def withCallback(cb: Instrument[F, Input] => F[Unit]): Self
```

Comment:
Ah, I see. I missed this part while making a draft.

We can mimic [Java API](https://github.com/open-telemetry/opentelemetry-java/blob/4a8850cc64cb407a33eb2131a2ef9cbc44567dcf/api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java) and add `ObservableMeasurement`:
```scala
trait ObservableMeasurement[A] {
  def record(value: A, attributes: attributes: Attribute[_]*): F[Unit]
}
```

Then, the `withCallback` can be:
```scala
def withCallback(cb: ObservableMeasurement[F, Input] => F[Unit]): Self
```